morebits.js

```diff
@@ -3563,6 +3563,41 @@ Morebits.wikitext.page.prototype = {
 	}
 };
 
+/**
+ * *********** Morebits.userspaceLogger ************
+ * Handles logging actions to a userspace log, used in
+ * twinklespeedy and twinkleprod.
+ */
+
+Morebits.userspaceLogger = function(logPageName) {
+	this.initialText = '';
+	this.headerLevel = 3;
+
+	this.log = function(logText, summaryText) {
+		var page = new Morebits.wiki.page('User:' + mw.config.get('wgUserName') + '/' + logPageName,
+			'Adding entry to userspace log'); // make this '... to ' + logPageName ?
+		return page.load(function(pageobj) {
+			var text = pageobj.getPageText();
```

Comment:
It took me a bit to wrap my head around these lines, since you unconditionally use getPageText and then conditionally reset the text if the page doesn't exist. This should work, but can I suggest
```
var text;
if (!pageobj.exists()) {
	text = this.initialText;
} else {
	text = pageobj.getPageText()
}
```

Reply:
I just copied from twinklespeedy.js and twinkleprod.js. But it's worth noting that `getPageText` returns an empty string if page doesn't exist.
so 
```js
var text = pageobj.getPageText();
if (!text) {
	text = this.initialText;
}
```
may be even better.